state/mongodb/mongodb.go

```diff
@@ -147,8 +151,14 @@ func (m *MongoDB) setInternal(ctx context.Context, req *state.SetRequest) error
 	}
 
 	// create a document based on request key and value
-	filter := bson.M{id: req.Key}
-	update := bson.M{"$set": bson.M{id: req.Key, value: vStr}}
+	var filter primitive.M
+	if req.ETag == nil || *req.ETag == "" {
```

Comment:
For 1.0.0-rc.3 we did a change that distinguishes between empty etag values and missing etag.

So this means that if req.ETag is not nil, it's value should be part of the etag.

Reply:
Well, well. We forgot this https://github.com/dapr/components-contrib/issues/731

Which means you get a string back and not *string.

So, I had to change the dapr/actors.go code to convert "" etag to nil etag when reading, so this would actually be nil. I can change this back but it is technically possible to happen in the future.